### PR TITLE
ACD-1351: Create errors CSV output for the import Xhibit

### PR DESCRIPTION
**Pull request by alexdesi**
Improves the XHIBIT cases import task to write failed rows to a CSV errors file.

- Adds CSV errors file output (`/tmp/import-errors.csv`) with all failed rows and their validation messages
- Makes `defendant_date_of_birth` optional with strict `yyyy-mm-dd` format validation when present
- Removes mandatory `defendant_arrest_summons_number` presence validation

### CODE REVIEW

File: lib/tasks/xhibit_cases.rake [hunk 1]
```diff
@@ -12,11 +14,23 @@ namespace :xhibit_cases do
...
+
+    errors_file = "/tmp/import-errors.csv"
+    input_headers = CSV.open(file_path, headers: true) { |csv| csv.first&.headers } || []
+
+    CSV.open(errors_file, "w") do |csv|
+      csv << input_headers + %w[errors]
+
+      results[:errors].each do |error|
+        csv << error[:row].values_at(*input_headers) + [error[:messages].join(" | ")]
+      end
```
Comment (sophiebrown01) [inline lib/tasks/xhibit_cases.rake]: I was going to suggest that we perhaps make this conditional based on there being any errors, but then I am also happy for us to create an empty errors CSV per run if there turn out to be no errors.

File: app/services/import_xhibit_cases.rb [hunk 2]
```diff
@@ -6,13 +6,13 @@ def initialize(file_path:)
...
-    CSV.foreach(file_path, headers: true).with_index(2) do |row, line|
+    results = { success_count: 0, errors: [] }
+    CSV.foreach(file_path, headers: true).with_index(2) do |row, line_number|
       xhibit_case = XhibitMigratedCase.create(row.to_h.transform_values(&:presence))
       if xhibit_case.persisted?
-        results[:inserted] << { line:, case_urn: xhibit_case.case_urn }
+        results[:success_count] += 1
       else
-        results[:errors] << { line:, case_urn: row["case_urn"], messages: xhibit_case.errors.full_messages }
+        results[:errors] << { line_number:, case_urn: row["case_urn"], row: row.to_h, messages: xhibit_case.errors.full_messages }
```
Comment (sophiebrown01) [inline app/services/import_xhibit_cases.rb:15]: Just a note that we might not have a case urn (as it's not mandatory)

**Comment by sophiebrown01** [comment]
I don't think I see it in this PR - we should also have a test to ensure that both the sent date and committal date are not empty. We also do not expect the sent date and committal date to _both_ be populated, this should be another error case. (so basically, the only valid condition is `sent_date XOR committal_date`

**Review by sophiebrown01 — APPROVED** [review]
Thanks for addressing those comments Alessandro, looks good! 🥇

File: lib/tasks/xhibit_cases.rake (hunk 1 above)
Comment (alexdesi) [reply lib/tasks/xhibit_cases.rake]: this made me think we can timestamp the filename, so we know when it's been run, 
being empty is proof that no errors were present for that time.

`/tmp/import-errors-1780583810.csv` for example.

**Comment by sophiebrown01** [comment]
I've done a check against the ACs for this task and I think there are a few missing tests, for:

- AC2: Duplicate URN
- AC9: Missing Case Type

I have referred back to the original DLRM requirements doc and it's worth noting that both case sub-type _and_ mode of trial are not mandatory. I can see in the original ACs that this distinction wasn't made, but if we receive a file without these fields then it is expected behaviour.

File: app/services/import_xhibit_cases.rb (hunk 2 above)
Comment (alexdesi) [reply app/services/import_xhibit_cases.rb:15]: Ok, I removed the `urn` mandatory, but we keep it the uniqueness (scoped within the defendant name and lastname)

**Comment by alexdesi** [comment]
> I've done a check against the ACs for this task and I think there are a few missing tests, for:
> 
>     * AC2: Duplicate URN
Test was already present. 
It was added in the previous PR, so probably you missed it:
https://github.com/ministryofjustice/laa-court-data-adaptor/blob/main/spec/models/xhibit_migrated_case_spec.rb#L27

After you comment, I've improved the error message to be more explicit.
> 
>     * AC9: Missing Case Type
test already present: https://github.com/ministryofjustice/laa-court-data-adaptor/pull/1528/changes#diff-8855db9e6477e81c9c214840a6047b0873eac6ad875bb495095043e41c5c6f1bR29

> 
> 
> I have referred back to the original DLRM requirements doc and it's worth noting that both case sub-type _and_ mode of trial are not mandatory. I can see in the original ACs that this distinction wasn't made, but if we receive a file without these fields then it is expected behaviour.

Yes, as you said, case sub-type and mode of trial are not mandatory.
I've updated the code.

File: spec/services/import_xhibit_cases_spec.rb [hunk 3]
```diff
@@ -17,8 +17,86 @@
...
+    end
+
+    it "reports a missing OU code as an error" do
+      result = import
+      error = result[:errors].find { |e| e[:case_urn] == "30GD0001003" }
+      expect(error).to include(line_number: 8, case_urn: "30GD0001003")
+      expect(error[:messages]).to include("Ou code can't be blank")
+    end
+
+    it "reports a missing case sub-type as an error" do
```
Comment (sophiebrown01) [inline spec/services/import_xhibit_cases_spec.rb]: I think this should be case type as case sub-type is optional

File: spec/services/import_xhibit_cases_spec.rb [hunk 4]
```diff
@@ -17,8 +17,86 @@
...
+      result = import
+      error = result[:errors].find { |e| e[:case_urn] == "30GD0001006" }
+      expect(error).to include(line_number: 11, case_urn: "30GD0001006")
+      expect(error[:messages]).to include("Case sub type can't be blank")
```
Comment (sophiebrown01) [inline spec/services/import_xhibit_cases_spec.rb]: And this changed to be case type too

File: spec/services/import_xhibit_cases_spec.rb (hunk 3 above)
Comment (alexdesi) [reply spec/services/import_xhibit_cases_spec.rb]: yes, this is not needed anymore (and actually this was failing)

File: spec/services/import_xhibit_cases_spec.rb (hunk 4 above)
Comment (alexdesi) [reply spec/services/import_xhibit_cases_spec.rb]: not needed, removed
